src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPostBuildStatus.java

```diff
@@ -43,6 +42,11 @@ public Revision decorateRevisionToBuild(GitSCM scm, Run<?, ?> run, GitClient git
             listener.getLogger().println("Injector could not be found while creating build status");
             return rev;
         }
+        // If the class was serialized before upgrading to 1.1.2
+        if (refNameExtractorFactory == null) {
+            refNameExtractorFactory = new BitbucketRefNameExtractorFactory();
+        }
+        String repositoryName = GitSCMUtil.getGitRepository(scm).getName();
```

Comment:
As mentioned in another comment below, I'd prefer to iterate over the repos and filter by `startsWith(branchName, repoName + "/")` and if there's a match then remove the prefix.
I think it's also good to break out of the loop as soon as we have a match, because we may be _really_ unlucky and something like this happens in the future:
repo1: `foo`
repo2: `bar`
(actual) ref name: `foo/blah`
branch name we get in the build info (i.e. `repo-name/ref`): `bar/foo/blah`

In this scenario, we may remove `bar/` from the beginning, and then iterating over the repos, also remove `foo/` and then oops, we've sent back the wrong ref name. :-)

Something like this is what I'm thinking:
```
String ref = ... refNameExtractorFactory.forBuildType(run.getClass())
                                    .extractRefName(branch.getName(), gitSCM.getRepositories().stream().map(/* get names /*)...);
```
And in the default extractor:
```
private static final class DefaultBitbucketRefNameExtractor implements BitbucketRefNameExtractor {

        @Override
        public String extractRefName(String branchSpec, Collection<String> repositoryNames) {
            String repositoryName = requireNonNull(repositoryNames, "repositoryNames").stream()
                                                     .filter(/* does the branchSpec start with this + '/'? */)
                                                     .findFirst()
                                                     .orElse("");
            return removeStart(branchSpec, requireNonNull(repositoryName, "repositoryName") + "/");
        }
    }
```

With the `.orElse("")` part, I'm conflicted whether we should ignore (basically what using empty string is), or throw an exception that the branchSpec doesn't start with any of the given repo names as expected. Anyone has any thoughts on this?

Reply:
I prefer empty String over throwing. It feels safer, and if we throw we can only log it anyway (we should not fail the build anyway). Empty string would mean that we don't modify the branch name, and most likely send the correct branch name. If git plugin fixes this "bug" (if they consider it as such) this code would continue to work without us rushing out a fix.

Reply:
I agree this is a potential edge we need to account for when we have multiple repos, but given we don't support that right now we can probably leave this question for then? I think there are a lot of places throughout our code where we assume one repo that will need to be updated in that case (to which this belong in that set). For now, unless the SCM has been incorrectly configured, there should be exactly 1 repo- but it's harmless to use `findFirst` so agree we should do that to account for that future issue.